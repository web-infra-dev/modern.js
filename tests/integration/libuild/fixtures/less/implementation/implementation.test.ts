import { getLibuilderTest } from '@/toolkit';

describe('fixture:less', function () {
  it('implemention object', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      input: {
        basic: './basic.less',
      },
      style: {
        less: {
          implementation: require('less'),
        },
      },
    });
    await bundler.build();
    bundler.expectCSSOutputMatchSnapshot();
  });
  it('implemention string', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      input: {
        basic: './basic.less',
      },
      style: {
        less: {
          implementation: require.resolve('less'),
        },
      },
    });
    await bundler.build();
    bundler.expectCSSOutputMatchSnapshot();
  });
});
