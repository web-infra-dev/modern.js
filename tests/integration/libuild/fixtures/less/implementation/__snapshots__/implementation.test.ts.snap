// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixture:less implemention object 1`] = `
"/* basic.less */
.alert {
  border: 1px solid rgba(198, 83, 140, 0.88);
}
"
`;

exports[`fixture:less implemention string 1`] = `
"/* basic.less */
.alert {
  border: 1px solid rgba(198, 83, 140, 0.88);
}
"
`;
